src/main/java/com/atlassian/db/replica/api/KeyBasedThrottledCache.java

```diff
@@ -0,0 +1,27 @@
+package com.atlassian.db.replica.api;
+
+import com.atlassian.db.replica.internal.util.ThreadSafe;
+import com.atlassian.db.replica.spi.SuppliedCache;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+/**
+ * Allows for thread-safe key-based access to ThrottledCache.
+ * It can be used to have only one instance per key (e.g. database url).
+ */
+@ThreadSafe
+public class KeyBasedThrottledCache implements SuppliedCache<Long> {
+    private static final ConcurrentHashMap<String, ThrottledCache<Long>> caches = new ConcurrentHashMap<>();
+    private final String key;
+
+    public KeyBasedThrottledCache(String key) {
+        this.key = key;
+        caches.putIfAbsent(key, new ThrottledCache<>());
```

Comment:
A new `ThrottledCache` will be created for every `KeyBasedThrottledCache`. Can you change it to `computeIfAbsent`?